### PR TITLE
[docs] Update PrereleaseNotice to support a callout

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Based on @Simek's feedback that both the callout and collapsible were serving the same purpose, in this PR:
- Updated the `PrereleaseNotice` component to use a default type of callout and next collapsible while still reusing the  component
- This helps us to not have two similar pieces of info in different contexts
- Also refactor the component
- Apply changes to Camera Next and SQLite Next references
- And apply prettier formatting where necessary in SQLite next code snippets

# Test Plan

Run docs locally.

## Preview

CleanShot 2024-02-19 at 18 35 51@2x

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/components/PrereleaseNotice.tsx [hunk 1]
```diff
@@ -1,17 +1,29 @@
-import ReactMarkdown from 'react-markdown';
-
-import { mdComponents } from './plugins/api/APISectionUtils';
+import { PropsWithChildren } from 'react';
 
+import { Callout } from '~/ui/components/Callout';
 import { Collapsible } from '~/ui/components/Collapsible';
+import { A, CODE } from '~/ui/components/Text';
+
+type Props = PropsWithChildren<object>;
```
Comment (Simek) [inline docs/components/PrereleaseNotice.tsx:7]:
```suggestion
type Props = PropsWithChildren;
```

Typing nit. We can also inline the type in that case, but was not able to suggest that.

**Review by Simek — APPROVED** [review]
Tested locally, thanks for working on the initial feedback idea further, LGTM! 👍

File: docs/components/PrereleaseNotice.tsx (hunk 1 above)
Comment (amandeepmittal) [reply docs/components/PrereleaseNotice.tsx:7]: Ah okay, i missed that. Let me update this. Thanks!

Comment (amandeepmittal) [reply docs/components/PrereleaseNotice.tsx]: Done ✅

**Pull request merged**